os-ui/src/pages/ChooseCategoryPage/ChooseCategoryPage.tsx

```diff
@@ -0,0 +1,81 @@
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../redux/Store';
+import NavbarOnSignIn from '../../component/NavbarOnSignIn/NavbarOnSignIn';
+import Search from '../../component/Search/Search';
+import Loader from '../../component/Loader/Loader';
+import { getSearchedCategories } from '../../services/getSearchedCategories';
+import { savePreferredCategories } from '../../services/savePreferredCategories';
+import {
+  CHOOSE_CATEGORIES_HEADER,
+  GET_CATEGORY_SUBCATEGORY_SEARCH_URL,
+  SAVE_PREFERRED_CATEGORIES,
+  EMPTY_DATA_ERROR_MESSAGE,
+} from '../../constants/Strings';
+import { SearchedCategoryType } from '../../types/SearchedCategoryType';
+import CategoryWithSubcategoriesProfile from '../../component/CategoryWithSubcategoriesProfile/CategoryWithSubcategoriesProfile';
+import styles from './ChooseCategoryPage.module.scss';
+
+const ChooseCategoryPage = () => {
+  const userInfo = useSelector<RootState>((state) => state.userInfo);
+  const subcategoryIdArray = useSelector<RootState>((state) => state.chooseSubcategories);
+  const navigate = useNavigate();
+  const [title, setTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [searchedCategories, setSearchedCategories] = useState<SearchedCategoryType>({});
+  const [isLoading, setIsLoading] = useState(true);
+  const { mainHeader, categoriesList, nextButton } = styles;
+
+  const handleChangeUrlTitleParam = (titleParam:string) => {
+    setTitle(titleParam);
+  };
+  const handleSavingCategories = () => {
+    savePreferredCategories(
```

Comment:
please try to separate so called service components where you can do calls to the back, in page components just *import these services.